packages/odata-v4/src/de-serializers/converters.ts

```diff
@@ -4,7 +4,7 @@ import moment from 'moment';
  * @internal
  */
 export function deserializeDateToMoment(date: string): moment.Moment {
-  const parsed = moment.utc(date, 'YYYY-MM-DD', true);
+  const parsed = moment.utc(date, 'Y-MM-DD', true);
```

Comment:
[q] The single `Y` means any number of years?

Reply:
Yes this means any number interpreted as years. This is the only way that I could find that moment parses >4-digit and negative dates (which are both supported by the spec).

Do you think the error message should be changed or is it better to ignore the edge-cases in the error message for clarity?

Reply:
Better ignore it. In 99% of the cases it will be 2020 or so.